Sources/Orbit/Components/Slider.swift

```diff
@@ -0,0 +1,441 @@
+                            dragStartOffset = value.startLocation.x - horizontalPosition
+                            isDragging.wrappedValue = true
+                        }
+
+                        let newHorizontalPosition = value.location.x - dragStartOffset
+
+                        guard newHorizontalPosition < width else {
+                            onValueChange(bounds.upperBound)
+                            return
+                        }
+
+                        guard newHorizontalPosition > 0 else {
+                            onValueChange(bounds.lowerBound)
+                            return
+                        }
+
+                        if buckets.count > 1 {
+                            let currentValue = currentValue(location: newHorizontalPosition, width: width)
+                            guard let bucket = buckets.first(where: { $0.contains(currentValue) }) else { return }
+                            onValueChange(currentValue < bucket.middle ? bucket.lowerBound : bucket.upperBound)
+                        } else {
+                            onValueChange(currentValue(location: newHorizontalPosition, width: width))
+                        }
+                    }
+                    .onEnded { _ in
+                        isDragging.wrappedValue = false
+                        dragStartOffset = 0
+                    }
+            )
+    }
+
+    @ViewBuilder func strokedPath(from: CGFloat, to: CGFloat) -> some View {
+        Path { path in
+            path.move(to: .init(x: from, y: lineCenter))
+            path.addLine(to: .init(x: to, y: lineCenter))
+        }
+        .stroke(Color.blueNormal, lineWidth: lineHeight)
+    }
+
+    private var lineHeight: CGFloat {
+        CGFloat.xxSmall * sizeCategory.controlRatio
+    }
+
+    private var lineCenter: CGFloat {
+        lineHeight / 2
+    }
+
+    private var thumbVerticalOverlapping: CGFloat {
+        (SliderThumb.circleDiameter / 2 - lineCenter) * sizeCategory.controlRatio
+    }
+
+    private var accessibilityHint: String {
+        "\(startLabel) - \(endLabel)"
+    }
+
+    private func updateSelectedRangeIfNeeded() {
+        if let upperBound = secondValue, case .range(let selectedRange) = value {
+            selectedRange?.wrappedValue = min(firstValue, upperBound)...max(firstValue, upperBound)
+        }
+    }
+
+    private func currentValue(location: CGFloat, width: CGFloat) -> Value {
+        bounds.lowerBound + Value(location / width) * bounds.distance
+    }
+
+    private func position(by value: Value, width: CGFloat) -> CGFloat {
+        let position = CGFloat((value - bounds.lowerBound) / bounds.distance) * width
+        return min(width, max(0, position))
+    }
+
+    public init(
+        value: Binding<Value>,
+        valueLabel: String,
+        startLabel: String,
+        endLabel: String,
+        bounds: ClosedRange<Value>,
+        step: Value.Stride? = nil
```

Comment:
If possible, the step value could be used for rounding the value so that call site does not have to clean it:

<img width="624" alt="image" src="https://user-images.githubusercontent.com/35844477/234886377-d3910a37-0636-4598-b923-2526792374b2.png">

Reply:
`Step` is an optional parameter, there is no default value 🤷 

Looks like I can't find a way how to fix this. The SwiftUI slider has the same issue and this issue is enevitable with working with doubles. So I would leave it up to client. Working with BinaryFloatingPoint doesn't allow much rounding either. We can round to  Int  but not to some fraciton

I've tried to use a Decimal type instead of BinaryFloatingPoint. This resolved issue with buckets (the print output):
```swift
[
Range(0..<10.3), Range(10.3..<20.6), Range(20.6..<30.9), Range(30.9..<41.2), Range(41.2..<51.5), 
Range(51.5..<61.8), Range(61.8..<72.1), Range(72.1..<82.4), Range(82.4..<92.7), Range(92.7..<100)
]
```
However converting from Decimal to Int/Double/Float will be more tricker. I actually don't even know the way except of using obj-C NSDecimalNumber for that 🤷 

So decimal gives us more precision but will introduce complications with actual component usage